### PR TITLE
Add commands to console logs page

### LINKED ISSUE #6626: Run Resource Commands from Console Logs Page
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I've been doing some iteration of onboarding some db migrations into Aspire, which has involved me 

1. Go to Dashboard
2. Go to Console Logs
3. See Error and tweak migrations scripts to fix
4. Go back to the Dashboard to restart the resource
5. Go back to Console Logs to see why the process errored this time

As I'm able to iterate on the migrations outside of aspire, I can test changes by simply restarting the migration resoruce, rather than rebuilding the whole app host.

### Describe the solution you'd like

I'd like the ability to be able to start / stop and restart a resource (and potentially any other resource command) directly from the Console Logs page, without having to go back to to the dashboard and then back to console logs.  That way I could stay on the console log page the entire time - view errors, fix them then restart the migrations from within the console logs page.

Something like:
Image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JamesNK**
## Description

Fixes https://github.com/dotnet/aspire/issues/6626

Demo:

consolelogs-commands

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6819)

### CODE REVIEW

**Comment by JamesNK** [comment]
Feels good to use. cc @davidfowl @DamianEdwards @leslierichardson95 in case you have any UI feedback.

**Comment by davidfowl** [comment]
This looks really good to me

**Comment by drewnoakes** [comment]
I wonder if some users will expect this to stop the logs, rather than the resource, which could be annoying or even dangerous (in ACA for example).

Some ideas that might help with this:

- Use a button with icon and text
- Move the start/stop button into the ... menu, so it always appears with text
- Move buttons next to resource combo box

**Comment by JamesNK** [comment]
The button icons and text come from the resource commands. Right now they just say `Stop`, `Start`, `Restart`. The text doesn't make it any clearer what the button does.

The buttons on the page could be removed and commands are only in a nested `Resource commands` menu.

e.g. to stop a resource, you would go `...` -> `Resource commands` -> `Stop`

**Comment by davidfowl** [comment]
If we add log specific commands where would they go?

**Comment by JamesNK** [comment]
In the example above? Resource commands

**Comment by JamesNK** [comment]
The tooltip can say what the button does:

image

Is that enough to remove confusion?

**Comment by drewnoakes** [comment]
Only if the user reads the tool tip. Having a button that stops a resource without any confirmation that might be mistaken to do something else seems like it could cause some undue problems.

I'm thinking of the scenario where something is logging rapidly and the user wants to pause the logs to make it easier to read them, so they click the "stop" button thinking that's what it does.

What about splitting that "toolbar" area such that all resource related stuff is on one side (combo box, commands) and all log related stuff is on the other side (status, future pause/resume/clear buttons).

**Comment by JamesNK** [comment]
I don't see a good place to put command buttons on the left. If they were placed between the dropdown and status message it wouldn't look good because there can be a variable number of commands for a resource, and the status message location would jump around.

**Comment by JamesNK** [comment]
How about this:

image

**Comment by drewnoakes** [comment]
> How about this:
> 
> image

This is what I had in mind. Seeing it, I think it's much clearer.

File: src/Aspire.Hosting/ApplicationModel/CommandsConfigurationExtensions.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ internal static void AddLifeCycleCommands(this IResource resource)
                     return ResourceCommandState.Hidden;
                 }
             },
-            displayDescription: null,
+            displayDescription: "Start resource",
```
Comment (drewnoakes) [inline src/Aspire.Hosting/ApplicationModel/CommandsConfigurationExtensions.cs:47]: Any ideas on how we could localize these? Options including having the server send all translated values, or sending a known key value that's then looked up in the dashboard.

**Review by drewnoakes — APPROVED** [review]

File: src/Aspire.Hosting/ApplicationModel/CommandsConfigurationExtensions.cs (hunk 1 above)
Comment (JamesNK) [reply src/Aspire.Hosting/ApplicationModel/CommandsConfigurationExtensions.cs:47]: I don't think there is any plan for this kind of text to be localizable. Like state text, the API for adding commands just takes one string. That would need to change, and then I guess the server would need to send the frontend lang in a cookie or query string.

**Pull request merged**